deepdrr/vol.py

```diff
@@ -159,11 +160,139 @@ def from_nifti(
 
     @classmethod
     def from_dicom(
-        cls,
-        path: Union[str, Path],
-    ) -> Volume:
-        """Create the volume from a DICOM file."""
-        raise NotImplementedError('load a volume from a dicom file')
+            cls,
+            path: Path,
+            use_thresholding: bool = True,
+            world_from_anatomical: Optional[geo.FrameTransform] = None,
+            use_cached: bool = True,
+            cache_dir: Optional[Path] = None
+    ):
+        """
+        load a volume from a dicom file and compute the anatomical_from_ijk transform from metadata
+        https://www.slicer.org/wiki/Coordinate_systems
+        Args:
+            path: path-like to a multi-frame dicom file. (Currently only Multi-Frame from Siemens supported)
+            use_thresholding (bool, optional): segment the materials using thresholding (faster but less accurate). Defaults to True.
+            world_from_anatomical (Optional[geo.FrameTransform], optional): position the volume in world space. If None, uses identity. Defaults to None.
+            use_cached (bool, optional): [description]. Use a cached segmentation if available. Defaults to True.
+            cache_dir (Optional[Path], optional): Where to load/save the cached segmentation. If None, use the parent dir of `path`. Defaults to None.
+
+        Returns:
+            Volume: an instance of a deepdrr volume
+        """
+        path = Path(path)
+        stem = path.name.split('.')[0]
+
+        if cache_dir is None:
+            cache_dir = path.parent
+
+        # Multi-frame dicoms store all slices of a volume in one file.
+        # they must specify the necessary dicom tags under
+        # https://dicom.innolitics.com/ciods/enhanced-ct-image/enhanced-ct-image-multi-frame-functional-groups
+        assert path.is_file(), 'Currently only multi-frame dicoms are supported. Path must refer to a file.'
+        logger.info(f'loading Dicom volume from {path}')
+
+        # reading the dicom dataset object
+        ds = dcmread(path)
+
+        # extracting all needed tags TODO add try exepts
+        frames = ds.PerFrameFunctionalGroupsSequence
+        shared = ds.SharedFunctionalGroupsSequence[0]
+        num_slices = len(frames)
+        first_slice_position = np.array(frames[0].PlanePositionSequence[0].ImagePositionPatient)
+        last_slice_position = np.array(frames[-1].PlanePositionSequence[0].ImagePositionPatient)
```

Comment:
@benjamindkilleen How should we handle the case that the supplied path is not a valid dicom? Atm, pydicom will throw some errors which the user should be able to decipher. Any ideas?

Reply:
I think anyone using DeepDRR should be familiar enough with pydicom that their error handling is sufficient. The option is to catch the error and suggest something more helpful, but I don't think that's needed. I saw you imported the InvalidDicomError?

Reply:
I agree, the error handling by pydicom should be sufficient. I imported the InvalidDicomError in an effort to implement some custom error handling.

I will remove the import now.